includes/wikia/services/ArticleVideoService.class.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\ArticleVideo\Api\MappingsInternalApi;
+use Swagger\Client\ArticleVideo\Models\Mapping;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Logger\WikiaLogger;
+use Wikia\Service\Swagger\ApiProvider;
+
+/**
+ * Created by PhpStorm.
+ * User: ryba
+ * Date: 11/12/2017
+ * Time: 12:24
+ */
+class ArticleVideoService {
+	const SERVICE_NAME = 'article-video';
+	const INTERNAL_REQUEST_HEADER = 'X-Wikia-Internal-Request';
+
+	private static $articleVideoApiClient = null;
+
+	/**
+	 * @param $cityId
+	 *
+	 * @return Mapping[]
+	 * @internal param $pageId
+	 *
+	 */
+	public static function getFeaturedVideosForWiki( string $cityId ): array {
+		$key = wfMemcKey( 'article-video', 'get-for-product', $cityId );
+
+		return WikiaDataAccess::cache(
+			$key,
+			WikiaResponse::CACHE_SHORT,
+			function () use ( $cityId ) {
+				$api = self::getMappingsInternalApiClient();
+				$api->getApiClient()->getConfig()->setApiKey( self::INTERNAL_REQUEST_HEADER, '1' );
+				$mappings = [];
+
+				try {
+					list( $response, $code ) = $api->getForProductWithHttpInfo( $cityId );
+
+					if ( $code == 200 ) {
+						$mappings = $response;
+					}
+				} catch ( ApiException $apiException ) {
+					WikiaLogger::instance()->debug(
+						'could not fetch data from article-video service',
+						[
+							'exception' => $apiException,
+							'status_code' => intval( $apiException->getCode() )
+						]
+					);
+				}
+
+				return $mappings;
```

Comment:
is there possibility to return current cached value instead of empty array when api call fails?

Reply:
I had an idea to do that, but it would require to use second layer of caching and as I discussed it with @hakubo it could be painful when we extend API of the service somewhere in the future.